docs/key-manager/key-manager.md

```diff
@@ -0,0 +1,121 @@
+---
+id: key-manager
+title: Key Manager
+---
+
+
+### Key Manager
+
+The key manager module's goal is to encapusulate all private key interactions with the [wallet](wallet.md) project.
+
+As of this writing, the wallet just delegates storage of the encrypted mnemonic seed to the key manager project. Over the log run, we want to make it so that the wallet project needs to communicate with the key-manager to access private keys.
+
+This means that ALL SIGNING should be done inside of the key-manager, and private keys should not leave the key manager.
+
+This makes it easier to reason about the security characteristics of our private keys, and a way to provide a uniform interface for alternative key storage systems (hsm, cloud based key storage, etc) to be plugged into the bitcoin-s library.
+
+#### Creating a key manager
+
+To create a key manager the first thing you need create a key manager is some entropy. A popular way for bitcoin wallet's to represent entropy is [BIP39](https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki)
+
+You can generate a `MnemonicCode` in bitcoin-s with the following code
+
+```scala mdoc
+import org.bitcoins.core.crypto._
+
+//get 256 bits of random entropy
+val entropy = MnemonicCode.getEntropy256Bits
+
+val mnemonic = MnemonicCode.fromEntropy(entropy)
+
+//you can prinhatt the mnemonic seed with this
+println(mnemonic.words)
+```
+
+Now that we have a `MnemonicCode` that was securely generated, we need to now create `KeyManagerParams` which tells us how to generate
+generate specific kinds of addresses for wallets.
+
+`KeyManagerParams` takes 3 parameters:
+
+1. `seedPath` there is where we store the `MnemonicCode` on your file system
+2. [`purpose`](https://github.com/bitcoin-s/bitcoin-s/blob/master/core/src/main/scala/org/bitcoins/core/hd/HDPurpose.scala) which represents what type of utxo this `KeyManager` is associated with
```

Comment:
Should probably reference `BIP 43` rather than scala code

Reply:
So I like referencing our actual implementation for things in this documentation as it hopefully helps people find where things are at. I did add the BIP43 specific to the end of this sentenced though.